tests/phpunit/CRM/Utils/TokenConsistencyTest.php

```diff
@@ -506,8 +506,10 @@ public function testMembershipTokenConsistency(): void {
     ]);
     $tokens = $tokenProcessor->listTokens();
     // Add in custom tokens as token processor supports these.
-    $expectedTokens = array_merge($expectedTokens, $this->getTokensAdvertisedByTokenProcessorButNotLegacy());
-    $this->assertEquals(array_merge($expectedTokens, $this->getDomainTokens()), $tokens);
+    $expectedTokens = array_merge($expectedTokens, $this->getTokensAdvertisedByTokenProcessorButNotLegacy(), $this->getContributionRecurTokens());
+    // $expectedTokens contains unadvertised tokens (ie. won't show for selection for the user).
+    // But listTokens() only gets us the "advertised" tokens so we need to "remove" all the unadvertised ones before checking they are equal.
```

Comment:
@mattwire so what were the 'unadvertised' ones?

Reply:
I think there were some legacy contributionrecur ones that still work but are not consistent with naming conventions for "standard" tokenprocessor. When we call `getContributionRecurTokens()` we get all possible tokens for recur I think but we'd only be testing against a list of tokens advertised by token processor to test against. Basically if you removed `getUnadvertisedTokens()` from the assertEquals then the test will throw an error containing that list of "unadvertised tokens".

Reply:
Most of that legacy token stuff was transitional - ie during transition we wanted to be sure the new functions differed from the old functions in logical ways.

With https://github.com/civicrm/civicrm-core/pull/25052 merged we can start to remove calls to the old functions in the tests & add noisy deprecation to them